content/courses/native-onchain-development/paging-ordering-filtering-data-frontend.md

```diff
@@ -147,21 +180,32 @@ For example, you could search through a list of contacts by including a `memcmp`
 filter:
 
 ```tsx
+const DATA_OFFSET = 2; // The first 2 bytes of metadata they store versioning information and are not needed for the search.
+const DATA_LENGTH = 18; // Retrieve 18 bytes of relevant data - This length includes the specific part of the account's data that holds the necessary information for comparison.
 async function fetchMatchingContactAccounts(
   connection: web3.Connection,
   search: string,
 ): Promise<(web3.AccountInfo<Buffer> | null)[]> {
-  const accounts = await connection.getProgramAccounts(programId, {
-    dataSlice: { offset: 0, length: 0 },
-    filters: [
-      {
-        memcmp: {
-          offset: 13,
-          bytes: bs58.encode(Buffer.from(search)),
-        },
-      },
-    ],
-  });
+  const accounts = (await connection.getProgramAccounts(
+    new PublicKey(MOVIE_REVIEW_PROGRAM_ID),
+    {
+      dataSlice: { offset: DATA_OFFSET, length: DATA_LENGTH },
+      filters:
+        search === ""
+          ? []
+          : [
+              {
+                memcmp: {
+                  offset: 6, // Skip the first 6 bytes of metadata as they contain irrelevant metadata, starting comparison at the 7th byte where the searchable content begins.
```

Comment:
Please be specific. 'Irrelevant information' - what specifically? Eg, 

```
the first 6 bytes are used to store (something), so we can start our comparison afterwards
```

Reply:
fixed✅